stacks-signer/src/cli.rs

```diff
@@ -170,11 +173,34 @@ pub struct GenerateFilesArgs {
     pub timeout: Option<u64>,
 }
 
+#[derive(Parser, Debug, Clone)]
+/// Arguments for the generate-stacking-signature command
+pub struct GenerateStackingSignatureArgs {
+    /// BTC address used to receive rewards
+    #[arg(short, long, value_parser = parse_pox_addr)]
+    pub pox_address: PoxAddress,
+    /// The reward cycle to be used in the signature's message hash
+    #[arg(short, long)]
+    pub reward_cycle: u64,
+    /// Path to config file
+    #[arg(long, value_name = "FILE")]
+    pub config: PathBuf,
+}
+
 /// Parse the contract ID
 fn parse_contract(contract: &str) -> Result<QualifiedContractIdentifier, String> {
     QualifiedContractIdentifier::parse(contract).map_err(|e| format!("Invalid contract: {}", e))
 }
 
+/// Parse a BTC address argument and return a `PoxAddress`
+pub fn parse_pox_addr(pox_address_literal: &str) -> Result<PoxAddress, String> {
+    if let Some(pox_address) = PoxAddress::from_b58(pox_address_literal) {
```

Comment:
@hstove the `pox_address` can be bech32/segwit right? https://github.com/stacks-network/stacks-core/blob/c659fca2f3f674309139dfc5b97379d2dacc5065/stackslib/src/chainstate/stacks/address.rs#L71-L85

Reply:
Yep - added some tests

Reply:
Ah that function supports all addr types, `b58` name threw me off